blocks/math/include/gnuradio-4.0/math/Math.hpp

```diff
@@ -115,6 +125,221 @@ using Multiply = MathOpMultiPortImpl<T, std::multiplies<T>>;
 template<typename T>
 using Divide = MathOpMultiPortImpl<T, std::divides<T>>;
 
+template<typename T>
+using And = MathOpMultiPortImpl<T, std::bit_and<T>>;
+template<typename T>
+using Or = MathOpMultiPortImpl<T, std::bit_or<T>>;
+template<typename T>
+using Xor = MathOpMultiPortImpl<T, std::bit_xor<T>>;
+
+template<class T = void>
+struct max {
+    constexpr T operator()(const T& lhs, const T& rhs) const { return std::max(lhs, rhs); }
+};
+
+template<class T = void>
+struct min {
+    constexpr T operator()(const T& lhs, const T& rhs) const { return std::min(lhs, rhs); }
+};
+
+template<typename T>
+using Max = MathOpMultiPortImpl<T, max<T>>;
+
+template<typename T>
+using Min = MathOpMultiPortImpl<T, min<T>>;
+
+GR_REGISTER_BLOCK("gr::blocks::math::Negate", gr::blocks::math::MathOpImpl, ([T], std::negate<[T]>), [ uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double, gr::UncertainValue<float>, gr::UncertainValue<double>, std::complex<float>, std::complex<double> ])
+GR_REGISTER_BLOCK("gr::blocks::math::Not", gr::blocks::math::MathOpImpl, ([T], std::bit_not<[T]>), [ uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double, gr::UncertainValue<float>, gr::UncertainValue<double>, std::complex<float>, std::complex<double> ])
+GR_REGISTER_BLOCK("gr::blocks::math::Abs", gr::blocks::math::MathOpImpl, ([T], std::abs_op<[T]>), [ uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, double ])
```

Comment:
I know that the GR 3 block doesn't work for complex types, but your `abs_op` function below does.  I would complex types in.